test_requirements.txt

```diff
@@ -1,4 +1,5 @@
-pytest >= 6.0.1
-pytest-mock >= 3.2.0
-pytest-cases >= 2.1.2
-coverage >= 5.2.1
\ No newline at end of file
+pytest >= 6.1.1
+pytest-mock >= 3.3.1
+pytest-cases >= 2.3.0
+coverage >= 5.3
+parameterized >= 0.7.4
```

Comment:
Rather then adding a new dependency, can `@pytest.mark.parametrize` be used to do the same thing?

Reply:
It can, but it has been requested specifically to avoid using `pytest` in tests directly in order to allow people to use the testing platform of their choosing.
I personally prefer using `pytest` at any time, but I respect the design decisions of this project.

Reply:
My main pytest dislike is it's magical patching of thing so I guess the adding of a dep to allow both testing systems to be used works. Especially when this refactor is a big win imo.